--- a/Scripts/Misc/SkillCheck.cs
+++ b/Scripts/Misc/SkillCheck.cs
@@ -447,7 +447,8 @@
 
 				if (!from.Player || (Server.HavenPrototype.HavenPlayerCaps.IsFree(from,skill) || Server.HavenPrototype.HavenPlayerCaps.Counted(skills) + toGain <= skills.Cap))
 				{
-					skill.BaseFixedPoint = Math.Min(skill.CapFixedPoint, skill.BaseFixedPoint + toGain);
+					toGain = Server.HavenPrototype.HavenCompanionProgression.Amount(from, skill, toGain);
+                    skill.BaseFixedPoint = Math.Min(skill.CapFixedPoint, skill.BaseFixedPoint + toGain);
 
 					EventSink.InvokeSkillGain(new SkillGainEventArgs(from, skill, toGain));
 
