--- a/Scripts/Services/Pet Training/PetTrainingHelper.cs
+++ b/Scripts/Services/Pet Training/PetTrainingHelper.cs
@@ -1367,7 +1367,7 @@
 
         public static bool ValidateTrainingPoint(BaseCreature bc, SkillName skill)
         {
-            return bc.Skills[skill].Base > 0;
+            return bc.Skills[skill].Base > 0 || (Server.HavenPrototype.HavenPreview.Enabled && bc is Server.HavenPrototype.HavenSnowBear && (MagicSkills.Contains(skill) || CombatSkills.Contains(skill)));
         }
 
         public static bool CheckSecondarySkill(BaseCreature bc, SkillName skill)
