diff --git a/Projects/UOContent/Mobiles/PlayerMobile.cs b/Projects/UOContent/Mobiles/PlayerMobile.cs
index 83c0e4cca..e37de6406 100644
--- a/Projects/UOContent/Mobiles/PlayerMobile.cs
+++ b/Projects/UOContent/Mobiles/PlayerMobile.cs
@@ -326,7 +326,7 @@ public TimeSpan NextTailorBulkOrder
 
         public List<Mobile> PermaFlags { get; private set; }
 
-        public override int Luck => AosAttributes.GetValue(this, AosAttribute.Luck);
+        public override int Luck => AosAttributes.GetValue(this, AosAttribute.Luck) + Server.UOOffline.HavenNewcomerLuck.GetBonus(this);
 
         public BOBFilter BOBFilter { get; private set; }
 
@@ -2256,6 +2256,7 @@ public override bool CheckNonlocalDrop(Mobile from, Item item, Item target)
 
         protected override void OnLocationChange(Point3D oldLocation)
         {
+            Server.UOOffline.HavenNewcomerLuck.OnMoved(this, Map, oldLocation);
             CheckLightLevels(false);
 
             DuelContext?.OnLocationChanged(this);
@@ -2311,6 +2312,7 @@ public override bool CheckShove(Mobile shoved) =>
 
         protected override void OnMapChange(Map oldMap)
         {
+            Server.UOOffline.HavenNewcomerLuck.OnMoved(this, oldMap, Location);
             if (Map != Faction.Facet && oldMap == Faction.Facet || Map == Faction.Facet && oldMap != Faction.Facet)
             {
                 InvalidateProperties();
