diff --git a/Projects/UOContent/Multis/Houses/BaseHouse.cs b/Projects/UOContent/Multis/Houses/BaseHouse.cs
--- a/Projects/UOContent/Multis/Houses/BaseHouse.cs
+++ b/Projects/UOContent/Multis/Houses/BaseHouse.cs
@@ -2501,7 +2501,7 @@
                 SecureLevel.CoOwners => IsCoOwner(m),
                 SecureLevel.Friends  => IsFriend(m),
                 SecureLevel.Anyone   => true,
-                SecureLevel.Guild    => IsGuildMember(m),
+                SecureLevel.Guild    => IsGuildMember(m) || this is Server.UOOffline.HavenGuildCastle company && company.CompanyAccess(m),
                 _                    => false
             };
         }
